forgerock-authenticator/src/main/java/org/forgerock/android/auth/FRAClient.java

```diff
@@ -162,8 +183,9 @@ public Account getAccount(@NonNull Mechanism mechanism) {
      * not be found or updated.
      * @param account The Account to update.
      * @return boolean as result of the operation
+     * @throws AccountLockException if account is locked
      */
-    public boolean updateAccount(@NonNull Account account) {
+    public boolean updateAccount(@NonNull Account account) throws AccountLockException {
```

Comment:
For updateAccount, lockAccount, unlockAccount, the interface return boolean, but at the same time throw AccountLockException, should we just use either one (Prefer AccountLockException), if exception throw, we consider failed, otherwise success. This will make developer easier to handle the error scenario.

Reply:
Good point, the boolean as return is good to have, specially for the "hybrid" apps. I can remove the throw and catch this exception.

Reply:
Done